plugins/producer_plugin/producer_plugin.cpp

```diff
@@ -1382,39 +1409,44 @@ void producer_plugin_impl::schedule_production_loop() {
-            fc_dlog(_log, "Scheduling Block Production on Exhausted Block #${num} immediately", ("num", chain.pending_block_state()->block_num));
+            fc_dlog(_log, "Scheduling Block Production on Exhausted Block #${num} immediately",
+                          ("num", chain.head_block_num()+1));
          } else {
             _timer.expires_at(epoch + boost::posix_time::microseconds(expect_time.time_since_epoch().count()));
-            fc_dlog(_log, "Scheduling Block Production on Exhausted Block #${num} at ${time}", ("num", chain.pending_block_state()->block_num)("time",expect_time));
+            fc_dlog(_log, "Scheduling Block Production on Exhausted Block #${num} at ${time}",
+                          ("num", chain.head_block_num()+1)("time",expect_time));
          }
       }
 
-      _timer.async_wait([&chain,weak_this,cid=++_timer_corelation_id](const boost::system::error_code& ec) {
-         auto self = weak_this.lock();
-         if (self && ec != boost::asio::error::operation_aborted && cid == self->_timer_corelation_id) {
-            // pending_block_state expected, but can't assert inside async_wait
-            auto block_num = chain.pending_block_state() ? chain.pending_block_state()->block_num : 0;
-            auto res = self->maybe_produce_block();
-            fc_dlog(_log, "Producing Block #${num} returned: ${res}", ("num", block_num)("res", res));
-         }
-      });
+      _timer.async_wait( app().get_priority_queue().wrap( priority::high,
+            [&chain,weak_this,cid=++_timer_corelation_id](const boost::system::error_code& ec) {
+               auto self = weak_this.lock();
+               if( self && ec != boost::asio::error::operation_aborted && cid == self->_timer_corelation_id ) {
+                  fc_dlog( _log, "Produce block timer running at ${time}", ("time", fc::time_point::now()) );
+                  // pending_block_state expected, but can't assert inside async_wait
+                  auto block_num = chain.is_building_block() ? chain.head_block_num() + 1 : 0;
+                  auto res = self->maybe_produce_block();
+                  fc_dlog( _log, "Producing Block #${num} returned: ${res}", ("num", block_num)( "res", res ) );
+               }
+            } ) );
    } else if (_pending_block_mode == pending_block_mode::speculating && !_producers.empty() && !production_disabled_by_policy()){
       fc_dlog(_log, "Specualtive Block Created; Scheduling Speculative/Production Change");
-      EOS_ASSERT( chain.pending_block_state(), missing_pending_block_state, "speculating without pending_block_state" );
-      const auto& pbs = chain.pending_block_state();
-      schedule_delayed_production_loop(weak_this, pbs->header.timestamp);
+      EOS_ASSERT( chain.is_building_block(), missing_pending_block_state, "speculating without pending_block_state" );
+      schedule_delayed_production_loop(weak_this, chain.pending_block_time());
```

Comment:
For consistence use `pending_block_time` from above.

Reply:
I'm not following. Which `pending_block_time` are you referring to?

Reply:
That was me looking at the diff wrong. Ignore.